src/main.js

```diff
@@ -323,8 +323,11 @@ class MiniGraphCard extends LitElement {
   }
 
   renderGraph() {
-    const ready = this.entity[0] && this.Graph[0]._history !== undefined;
-
+    const ready = (this.entity[0] && !this.Graph.some(
+      (element, index) => element._history === undefined
+      && this.config.entities[index].show_graph !== false,
+    ))
+    || this.config.show.loading_indicator !== true;
```

Comment:
```suggestion
    || this.config.show.loading_indicator === false:
```

Can we change this to avoid having to define the value in `show`? It might later bite our asses in the UI config. See https://github.com/kalkih/mini-graph-card/pull/1128#discussion_r1761831060.

Reply:
> Sorry for messing this up. :-)

No problem, these small things are a part of doing anything. "Who does not do anything - does not make mistakes".

> had been told, efficient code was a goal of this project, originally.

Totally agree with the the main point that the original code is efficient. Currently we are adding small things; hopefully this "loading indicator" is a minor thing as well. I only noticed a slow "rotation" on an old iPhone 5S - and if we can disable this "loading indicator" by a new option - then it will not affect anything.

Mostly I like this indicator because it is a common thing in HA:
history page:
![image](https://github.com/user-attachments/assets/9ab7fc61-ffb4-4f1e-8da6-b1b2b6514d59)

logbook:
![image](https://github.com/user-attachments/assets/a6f7d934-ac3a-4e2c-aa0b-85fbdbb1c1f6)

Although we have this in history-graph:
![image](https://github.com/user-attachments/assets/fd6ddba5-4f33-4a43-a22f-991e0c9ae372)
We can choose this way of course - then we we'll have to load a local translation of this "Loading state history..." sentence (in "src\translations\xxxxxx.json").

Reply:
> Can we change this to avoid having to define the value in show? It might later bite our asses in the UI config

1. Imho we should add this new option. Yes, it increases a complexity a bit - but then we will have a possibility to "lighten up" a look (i.e. w/o a "loading indicator").

2. As for the UI: honestly this is a less prioritized feature for me.

3. As for default value: agree we should avoid listing all options in `DEFAULT_SHOW` for simplicity.
So I will consider your proposal.
The expression `this.config.show.loading_indicator !== true` is `true` if the option is explicitly defined in user config as `false` or it is undefined (i.e. does not have a default value).
The expression `this.config.show.loading_indicator === false` is `true` if the option is explicitly defined in user config as `false`; if it is undefined - then the expression will be `false` (like a default value is `false`).
If we want the indicator to be displayed by default - then we should use my way `!== true`.
Otherwise - we should use your way `=== false`.
Please correct me if I made a mistake.

Reply:
Let me think aloud, as this creates logical knots in my brain...

We're setting
```
ready = involved_logic || not loading_indicator
```

Which reduces, while loading, to
```
ready = not loading_indicator
```

The icon will be displayed, if `ready==false`. We want to show the indicator by default, so we want `not loading_indicator` to be `false` by default:

|  loading_indicator | ready | indicator shown |
| -- | -- | -- |
| false | true | no |
| true | false | yes |
| undefined | false | yes |

So, we should use `loading_indicator === false`. 

Right? 🤔 

I agree with everything else your wrote.
* Keep the blue circle.
* Add the option.
* Remove the explicit default setting.
* (And also reduced options in UI only)

Reply:
Now I got knots myself)))
![image](https://github.com/user-attachments/assets/5d8462da-e118-48a2-9c78-03efe4bf37e2)
I honestly do not understand why `not undefined` = `false`...

Checked myself again, I made a mistake in my prev post:
![image](https://github.com/user-attachments/assets/066a3f28-fd06-40e0-a4ea-c262838ef265)
Based on this - yes, we should use `loading_indicator === false`.
Also I will remove a default value from DEFAULT_SHOW .

Reply:
Thanks, looks good! I'll test it when I have some minutes at home and merge it!